cli/flox/src/commands/publish.rs

```diff
@@ -139,12 +152,57 @@ impl Publish {
             _ => bail!("Unsupported environment type"),
         };
 
+        let base_nixpkgs_url = {
+            let base_catalog_info = flox
+                .catalog_client
+                .get_base_catalog_info()
+                .await
+                .context("could not get information about the base catalog")?;
+
+            match stability {
+                Some(stability) => {
+                    let make_error_message = || {
+                        let available_stabilities =
+                            base_catalog_info.available_stabilities().join(", ");
+                        formatdoc! {"
+                          Stability '{stability}' does not exist (or has not yet been populated).
+                          Available stabilities are: {available_stabilities}
+                      "}
+                    };
+
+                    let url = base_catalog_info
+                        .url_for_latest_page_with_stability(&stability)
+                        .with_context(make_error_message)?;
+
+                    debug!(%url, %stability, "using page from user provided stability");
+                    url
+                },
+                None => {
+                    let make_error_message = || {
+                        let available_stabilities =
+                            base_catalog_info.available_stabilities().join(", ");
+                        formatdoc! {"
+                          The default stability {} does not exist (or has not yet been populated).
+                          Available stabilities are: {available_stabilities}
+                      ", BaseCatalogInfo::DEFAULT_STABILITY}
+                    };
+
+                    let url = base_catalog_info
+                        .url_for_latest_page_with_default_stability()
+                        .with_context(make_error_message)?;
+
+                    debug!(%url, "using page from default stability flake");
```

Comment:
Should this still say flake? Build was changed but publish wasn't. Keeping the two implementations in sync might also be a good reason to pull it out to a shared function.
```suggestion
                    debug!(%url, "using page from default stability");
```